### PR TITLE
[all] Check for `VERCEL_` environment variables with `getPlatformEnv()`

### PR DESCRIPTION
**Pull request by styfle**
Added the following env vars, most are undocumented but its good to be consistent:

- `VERCEL_REGION`
- `VERCEL_DEBUG`
- `VERCEL_BUILDER_DEBUG`
- `VERCEL_TOKEN`
- `__VERCEL_SKIP_DEV_CMD`

I also updated the error code prefixes to remove `NOW_`.
There `code` isn't used anywhere, this is just to make it unique and a little shorter.

### CODE REVIEW

File: packages/now-static-build/test/fixtures/45-node-modules-bin-path/now.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
   ],
   "build": {
     "env": {
-      "NOW_BUILDER_DEBUG": "1"
+      "NOW_BUILDER_DEBUG": "1",
+      "VERCEL_BUILDER_DEBUG": "1"
```
Comment (TooTallNate) [inline packages/now-static-build/test/fixtures/45-node-modules-bin-path/now.json:15]: Why are both of these necessary?

File: test/lib/deployment/now-deploy.js [hunk 2]
```diff
@@ -161,14 +164,25 @@ async function fetchWithAuth(url, opts = {}) {
...
   }
+  if (VERCEL_TOKEN) {
+    console.log('Using VERCEL_TOKEN for test deployment');
+    return VERCEL_TOKEN;
+  }
   if (!ZEIT_TEAM_TOKEN || !ZEIT_REGISTRATION_URL) {
     throw new Error(
-      'Failed to create test deployment. Did you forget to set NOW_TOKEN?'
+      process.env.CI
+        ? 'Failed to create test deployment. This is expected for git forks. Please run tests locally.'
```
Comment (TooTallNate) [inline test/lib/deployment/now-deploy.js]:
```suggestion
        ? 'Failed to create test deployment. This is expected for 3rd party pull requests. Please run tests locally.'
```

File: test/lib/deployment/now-deploy.js [hunk 3]
```diff
@@ -161,14 +164,25 @@ async function fetchWithAuth(url, opts = {}) {
...
+        : 'Failed to create test deployment. Please set VERCEL_TOKEN environment variable and run again.'
```
Comment (TooTallNate) [inline test/lib/deployment/now-deploy.js]:
```suggestion
        : 'Failed to create test deployment. Please set `VERCEL_TOKEN` environment variable and run again.'
```

**Review by TooTallNate — APPROVED** [review]
In general, code LGTM. However, I would have extracted the conditional logic out to a `getPlatformEnv()` helper function so that the "NOW_" or "VERCEL_" logic is in a single location.

File: packages/now-static-build/test/fixtures/45-node-modules-bin-path/now.json (hunk 1 above)
Comment (styfle) [reply packages/now-static-build/test/fixtures/45-node-modules-bin-path/now.json:15]: I think we'll have to deploy `@vercel/build-utils` for this change to take affect on `@vercel/static-build` here but maybe my assumption was wrong. Let me try it.

**Comment by styfle** [comment]
@TooTallNate Thanks, I added `getPlatformEnv()` and updated with your other suggestions too 👍

File: packages/now-build-utils/src/debug.ts [hunk 4]
```diff
@@ -1,5 +1,7 @@
+import { getPlatformEnv } from './index';
```
Comment (TooTallNate) [inline packages/now-build-utils/src/debug.ts]:
```suggestion
import { getPlatformEnv } from './';
```

File: test/lib/deployment/now-deploy.js [hunk 5]
```diff
@@ -145,9 +146,10 @@ async function fetchWithAuth(url, opts = {}) {
     currentCount += 1;
     if (!token || currentCount === MAX_COUNT) {
       currentCount = 0;
-      if (process.env.NOW_TOKEN) {
+      if (process.env.VERCEL_TOKEN || process.env.NOW_TOKEN) {
         // used for health checks
         token = process.env.NOW_TOKEN;
+        token = process.env.VERCEL_TOKEN;
       } else {
```
Comment (TooTallNate) [inline test/lib/deployment/now-deploy.js]: This doesn't seem right…

```suggestion
      // used for health checks
      token = process.env.VERCEL_TOKEN || process.env.NOW_TOKEN;
      if (!token) {
```

**Review by TooTallNate — CHANGES_REQUESTED** [review]

**Review by TooTallNate — APPROVED** [review]

**Pull request merged**